src/borg/archive.py

```diff
@@ -795,7 +795,7 @@ def process_file(self, path, st, cache, ignore_inode=False):
             # there should be no information in the cache about special files processed in
             # read-special mode, but we better play safe as this was wrong in the past:
             path_hash = ids = None
-        first_run = not cache.files
+        first_run = not cache.files and cache.do_files
```

Comment:
well, this silences it by making it non-functional when cache.do_files is False.
can we keep it functional?